src/record/mod.rs

```diff
@@ -309,6 +281,19 @@ impl Header {
         bincode::serialized_size(&self).expect("calc record serialized size")
     }
 
+    pub(crate) fn set_offset_checksum(&mut self, blob_offset: u64, header_checksum: u32) {
+        self.blob_offset = blob_offset;
+        self.header_checksum = header_checksum;
```

Comment:
Please, add `debug_assert` that verifies, that checksum is correct and equal to `update_checksum` result